lib/rubyplot/artist/figure.rb

```diff
@@ -109,7 +109,7 @@ def show
         Rubyplot.backend.canvas_height = @height
         Rubyplot.backend.canvas_width = @width
         Rubyplot.backend.figure = self
-        Rubyplot.backend.init_output_device(nil, device: nil)
```

Comment:
This is leading to copy pasting of code in the `write` and `show` functions. Abstract these into another method.